sqlglot/dialects/nuodb.py

```diff
@@ -1,9 +1,35 @@
 from __future__ import annotations
 
-from sqlglot import exp, generator, parser, tokens
-from sqlglot.dialects.dialect import Dialect
+from sqlglot import exp, generator, parser, tokens, transforms
+from sqlglot.dialects.dialect import Dialect, no_properties_sql
 from sqlglot.tokens import Tokenizer, TokenType
 
+def _auto_increment_to_generated_by_default(expression: exp.Expression) -> exp.Expression:
+    auto = expression.find(exp.AutoIncrementColumnConstraint)
+    if auto:
+        expression = expression.copy()
+        expression.args["constraints"].remove(auto.parent)
+        kind = expression.args["kind"]
+        if auto is not None:
+            expression.append("constraints", exp.ColumnConstraint(kind=exp.GeneratedAsIdentityColumnConstraint(this=False)))
+
+    return expression
+
+def _remove_collate(expression: exp.Expression) -> exp.Expression:
+    column_constraint = expression.find(exp.ColumnConstraint)
+    if column_constraint:
+        expression = expression.copy()
+        collateProp = column_constraint.find(exp.CollateColumnConstraint)
+        charsetProp = column_constraint.find(exp.CharacterSetColumnConstraint)
+        if collateProp:
+            expression.args["kind"].replace(None)
+        if charsetProp:
+            # print(expression.args["kind"])
+            expression.args["kind"].replace(None)
```

Comment:
If either `collateProp` or `charsetProp` is present, the `kind` arg is removed. I'd suggest to refactor a bit to something like:
```py
        collateProp = column_constraint.find(exp.CollateColumnConstraint)
        charsetProp = column_constraint.find(exp.CharacterSetColumnConstraint)
        if collateProp or charsetProp:
           expression.args["kind"].replace(None)
```